cfme/fixtures/v2v_fixtures.py

```diff
@@ -0,0 +1,449 @@
+import itertools
+import json
+from collections import namedtuple
+
+import fauxfactory
+import pytest
+from riggerlib import recursive_update
+from widgetastic.utils import partial_match
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.fixtures.provider import setup_or_skip
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
+from cfme.v2v.infrastructure_mapping import InfrastructureMapping as InfraMapping
+
+
+FormDataVmObj = namedtuple("FormDataVmObj", ["infra_mapping_data", "vm_list"])
+
+
+@pytest.fixture(scope="module")
+def v2v_provider_setup(request, source_provider, provider):
+    """ Fixture to setup providers """
+    V2vProviders = namedtuple("V2vProviders", ["vmware_provider", "rhv_provider", "osp_provider"])
+    vmware_provider, rhv_provider = None, None
+    for provider in [source_provider, provider]:
+        if provider.one_of(VMwareProvider):
+            vmware_provider = provider
+            setup_or_skip(request, vmware_provider)
+        elif provider.one_of(RHEVMProvider):
+            rhv_provider = provider
+            setup_or_skip(request, rhv_provider)
+        elif provider.one_of(OpenStackProvider):
+            osp_provider = provider
+            setup_or_skip(request, osp_provider)
+        else:
+            pytest.skip("Provider {} is not a valid provider for v2v tests".format(provider.name))
+    v2v_providers = V2vProviders(vmware_provider=vmware_provider,
+                                 rhv_provider=rhv_provider,
+                                 osp_provider=osp_provider)
+    yield v2v_providers
+    v2v_providers.vmware_provider.delete_if_exists(cancel=False)
+    v2v_providers.rhv_provider.delete_if_exists(cancel=False)
+    v2v_providers.osp_provider.delete_if_exists(cancel=False)
+
+
+@pytest.fixture(scope="module")
+def add_host_creds(request, v2v_providers):
+    """Add credentials to VMware and RHV hosts."""
+    if len(v2v_providers) > 2:
+        pytest.skip(
+            "There are more than two providers in v2v_providers fixture,"
+            "which is invalid, skipping."
+        )
+    try:
+        vmware_provider = v2v_providers.vmware_provider
+        rhv_provider = v2v_providers.rhv_provider
+        vmware_hosts = vmware_provider.hosts.all()
+        try:
+            for host in vmware_hosts:
+                host_data, = [
+                    data for data in vmware_provider.data["hosts"] if data["name"] == host.name
```

Comment:
So, some things to consider that I think make this less than ideal:
1. If there are multiple matching hosts that are in the yaml and returned by `provider.hosts.all()`, this hits ValueError - is that really invalid? Do we want to just pick the first host in that case? Isn't there potential that we'll want to update credentials on _all_ the hosts available to the provider, since any could be used in the test?

2. The call to `host.update_credentials_rest()` could be make directly in the list comprehension, since we're not using that host data otherwise. 

3. The code duplication here could be eliminated by creating `vmware_hosts` and `rhv_hosts` outside of the try block, and then iterating over each. The same list comprehension is being used against both host lists.